### PR TITLE
Make sure to actually process recurring charges only for existing ord…

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
…ers and users.

## Description
Added some logic to avoid processing recurring charges for deleted orders, or for existing orders whose user has been deleted.
When the requested code for #940 will be implemented, the code of this PR would still make sense for edge cases like  orders/users manually removed from the db.

## How has this been tested?
existing and new unit tests. **haven't ran a manual test!)**

## Types of changes
Improvement/possible breaking change (but unit tests say it's fine)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests.
- [x] My code follows the LifterLMS Coding Standards.

### CODE REVIEW

File: includes/controllers/class.llms.controller.orders.php [hunk 1]
```diff
@@ -415,51 +416,75 @@ public function pending_cancel_order( $order ) {
...
 	 */
 	public function recurring_charge( $order_id ) {
 
-		$order   = new LLMS_Order( $order_id );
-		$gateway = $order->get_gateway();
+		$order = llms_get_post( $order_id );
 
-		// ensure the gateway is still installed & available
-		if ( ! is_wp_error( $gateway ) ) {
+		if ( ! ( $order && is_a( $order, 'LLMS_Order' ) ) ) {
```
Comment (thomasplevy) [inline includes/controllers/class.llms.controller.orders.php]: I think this is a confusing syntax. I'd rather ` if ( ! $order || ! is_a( $order, 'LLMS_Order  ) {`

File: includes/controllers/class.llms.controller.orders.php [hunk 2]
```diff
@@ -415,51 +416,75 @@ public function pending_cancel_order( $order ) {
...
 
-			if ( ! $gateway->supports( 'recurring_payments' ) ) {
+			do_action( 'llms_order_recurring_charge_order_error', $order_id, $this );
 
-				do_action( 'llms_order_recurring_charge_gateway_payments_disabled', $order_id, $gateway, $this );
+			llms_log( 'Recurring charge for order # ' . $order_id . ' could not be processed because the order does no longer exist.', 'recurring-payments' );
```
Comment (thomasplevy) [inline includes/controllers/class.llms.controller.orders.php]: Grammar checker: "because the order no longer exists."

**Comment by thomasplevy** [comment]
I started leaving comments on this but it's all pretty minor, I'm going to make the changes myself and merge it in. looks (mostly) good!

File: includes/controllers/class.llms.controller.orders.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/controllers/class.llms.controller.orders.php]: Mmm okay, I find it more easy for my brain: "I want both or nothing..." 😁

**Comment by thomasplevy** [comment]
Closing in favor of #947
